core/types/calls.cc

```diff
@@ -3052,6 +3052,19 @@ class Module_tripleEq : public IntrinsicMethod {
             res.returnType = rhs;
             return;
         }
+
+        auto rhsSym = Symbols::noClassOrModule();
+        if (isa_type<ClassType>(rhs)) {
+            rhsSym = cast_type_nonnull<ClassType>(rhs).symbol;
+        } else if (auto *app = cast_type<AppliedType>(rhs)) {
+            rhsSym = app->klass;
+        }
+
+        if (rhsSym.exists() && rhsSym.data(gs)->isClassOrModuleSealed() &&
+            rhsSym.data(gs)->sealedLocs(gs).size() == 1) {
```

Comment:
Is this guaranteed to have the same number of elements as there are sealed subclasses? What if the code is generated by a DSL pass and the locals are not unique? (Idk if this currently happens but it could). I can’t remember whether there’s not a better way to find this size.

Reply:
It was not returning the locations of the subclasses, but instead the locations where it's allowable to subclass the sealed class. I added a helper that answers this question explicitly instead, and a comment on `sealedLocs`.